core/src/main/java/hudson/Util.java

```diff
@@ -1927,4 +1928,32 @@ public static long daysElapsedSince(@NonNull Date date) {
     private static PathRemover newPathRemover(@NonNull PathRemover.PathChecker pathChecker) {
         return PathRemover.newFilteredRobustRemover(pathChecker, DELETION_RETRIES, GC_AFTER_FAILED_DELETE, WAIT_BETWEEN_DELETION_RETRIES);
     }
+
+    /**
+     * Returns Digest of input bytes using specified algorithm
+     */
+    @Restricted(NoExternalUse.class)
+    public static byte[] getDigestOf(@NonNull byte[] input, @NonNull String algorithm) {
+        if (input == null || algorithm == null) {
+            return null;
+        }
+        try {
+                MessageDigest messageDigest = MessageDigest.getInstance(algorithm);
+                messageDigest.update(input);
+                return messageDigest.digest();
+        } catch (NoSuchAlgorithmException noSuchAlgorithmException) {
+            throw new IllegalStateException(algorithm + " could not be instantiated, but is required to" +
```

Comment:
this is incorrect.  I can pass "fooBarWibble" as the algorithm and it is not required to be implemented.

If you want to be generic then you have to account for various issues and force callers to handle them and react appropriately, given this is only currently by one function (getHexOfSHA256DigestOf) I would say YAGNI.  

if you do want somethign generic here - then the rest of the code should probably be updated to use it.

Reply:
Modified not to make it generic considering the complexity of the changes. Made it specific to SHA-256